src/DssSpell.sol

```diff
@@ -75,334 +60,240 @@ contract DssSpellAction is DssAction {
+        // Move ownership of AllocatorVault & AllocatorBuffer to AllocatorProxy (SparkProxy)
+        // Add Allocator contracts to chainlog (ALLOCATOR_ROLES, ALLOCATOR_REGISTRY, ALLOCATOR_SPARK_A_VAULT, ALLOCATOR_SPARK_A_BUFFER, PIP_ALLOCATOR_SPARK_A)
+        // Add ALLOCATOR-SPARK-A ilk to IlkRegistry
+
+        // Set sharedInstance with the following parameters:
+        AllocatorSharedInstance memory allocatorSharedInstance = AllocatorSharedInstance({
+            // Set parameter sharedInstance.oracle: (Allocator Oracle: 0xc7B91C401C02B73CBdF424dFaaa60950d5040dB7)
+            oracle:   PIP_ALLOCATOR,
+            // Set parameter sharedInstance.roles: (AllocatorRoles: 0x9A865A710399cea85dbD9144b7a09C889e94E803)
+            roles:    ALLOCATOR_ROLES,
+            // Set parameter sharedInstance.registry: (AllocatorRegistry: 0xCdCFA95343DA7821fdD01dc4d0AeDA958051bB3B)
+            registry: ALLOCATOR_REGISTRY
+        });
+
+        // Set ilkInstance with the following parameters:
+        AllocatorIlkInstance memory allocatorIlkInstance = AllocatorIlkInstance({
+            // Set parameter ilkInstance.owner: (MCD_PAUSE_PROXY)
+            owner:  MCD_PAUSE_PROXY,
+            // Set parameter ilkInstance.vault: (AllocatorVault: 0x691a6c29e9e96dd897718305427Ad5D534db16BA)
+            vault:  ALLOCATOR_SPARK_VAULT,
+            // Set parameter ilkInstance.buffer: (AllocatorBuffer: 0xc395D150e71378B47A1b8E9de0c1a83b75a08324)
+            buffer: ALLOCATOR_SPARK_BUFFER
+        });
+
+        // Set cfg with the following parameters:
+        AllocatorIlkConfig memory allocatorIlkCfg = AllocatorIlkConfig({
+            // Set parameter cfg.ilk: (ALLOCATOR-SPARK-A)
+            ilk             : "ALLOCATOR-SPARK-A",
+            // Set parameter cfg.duty: (5.2% -> 1000000001607468111246255079)
+            duty            : FIVE_PT_TWO_PCT_RATE,
+            // Set parameter cfg.gap: (10,000,000)
+            maxLine         : 10_000_000 * RAD,
+            // Set parameter cfg.maxLine: (10,000,000)
+            gap             : 10_000_000 * RAD,
+            // Set parameter cfg.ttl: (86,400 seconds)
+            ttl             : 86_400 seconds,
+            // Set parameter cfg.allocatorProxy: 0x3300f198988e4C9C63F75dF86De36421f06af8c4
+            allocatorProxy  : SPARK_PROXY,
+            // Set parameter cfg.ilkRegistry: 0x5a464c28d19848f44199d003bef5ecc87d090f87
+            ilkRegistry     : ILK_REGISTRY
+        });
+
+        // Init shared components for Allocator System
+        AllocatorInit.initShared(dss, allocatorSharedInstance);
+
+        // Init Allocator ILK for Spark Subdao by calling
+        AllocatorInit.initIlk(dss, allocatorSharedInstance, allocatorIlkInstance, allocatorIlkCfg);
+
+
+        // ---------- Whitelist Spark ALM Proxy on the PSM ----------
+        // Forum: https://forum.sky.money/t/spell-contents-2024-10-31/25421/
+
+        DssLitePsmLike(LITE_PSM).kiss(SPARK_ALM_PROXY);
```

Comment:
`IPSMLike(MCD_LITE_PSM_USDC_A).kiss(almProxy: 0x1601843c5E9bC251A3272907010AFa41Fa18347E);` comment is missing